File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-7.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func goodNodes(root *TreeNode) int {
    if root == nil {
		return 0
	}

	counter := 0

	var dfs func(root *TreeNode, curMaxVal int)
	dfs = func(root *TreeNode, curMaxVal int) {
		if root == nil {
			return 
		}

		if root.Val >= curMaxVal {
			counter++
		}
		curMaxVal = max(curMaxVal, root.Val)

		dfs(root.Left, curMaxVal)
		dfs(root.Right, curMaxVal)
	}

	dfs(root, root.Val)

	return counter
}

File: Data Structures & Algorithms/same-binary-tree/submission-7.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p, q):
        def dfs(p, q):
            if p is None and q is None:
                return True

            if p is None or q is None:
                return False

            is_node_same = p.val == q.val

            return (
                is_node_same
                and dfs(p.left, q.left)
                and dfs(p.right, q.right)
            )

        return dfs(p, q)